nservicebus/messaging/databus.md

```diff
@@ -66,9 +66,45 @@ snippet:MessageWithLargePayloadUsingConvention
 
 ## DataBus attachments cleanup
 
-NServiceBus DataBus implementations currently behave differently with regard to cleanup of physical attachments used to transfer data properties. `FileShareDataBus` **does not** remove physical attachments once the message is gone. `AzureDataBus` **does** remove Azure storage blobs used for physical attachments once the message is gone.
+NServiceBus DataBus implementations currently behaves differently with regard to cleanup of physical attachments used to transfer data properties depending on the implementation used.
 
 
+### Reasons why attachments are not removed by default
+
+Automatically removing these attachments can cause problems in many situations. For example:
+
+* The supported data bus implementations do not participate in distributed transactions. If for some reason, the message handler throws an exception and the transaction rolls back, the delete operation on the attachment cannot be rolled back. Therefore, when the message is retried, the attachment will no longer be present causing additional problems.
+* The message can be deferred so that the file will be processed later. Removing the file after deferring the message, results in a message without the corresponding file.
+* Functional requirements might dictate the message to be available for a longer duration.
+* If the outbox feature in NServiceBus is enabled, the message will be removed from the incoming queue, but it might not have been processed yet. 
+
+
+### AzureDataBus Implementation
+
+AzureDataBus will **remove** the Azure storage blobs used for physical attachments after the message is processed if the `TimeToBeReceived` attribute is specified. When this attribute isn’t provided, the physical attachments will not be removed.  
```

Comment:
I am linking to that article in the cleanup stragegy. Will remove the word attribute. Replacing it with value instead to be generic.